pkg/server/ca/ca.go

```diff
@@ -294,9 +296,26 @@ func signX509SVID(td spiffeid.TrustDomain, x509CA *X509CA, params X509SVIDParams
 		return nil, err
 	}
 
-	// In case subject is provided use it
-	if params.Subject.String() != "" {
+	switch {
+	// If the caller provided a subject, use it as the base subject.
+	case params.Subject.String() != "":
 		template.Subject = params.Subject
+	// Otherwise, if the new behavior has been disabled, then use the old
+	// subject value
+	case omitUID:
+		template.Subject = pkix.Name{
+			Country:      []string{"US"},
+			Organization: []string{"SPIRE"},
+		}
+	// Otherwise, use the CA subject (sans the common name) as a base.
+	default:
+		template.Subject = x509CA.Certificate.Subject
```

Comment:
is it possible to have an empty Subject on CA? I think it is not possible, and our default will be "O=SPIFFE,C=US", if that is the case, we'll need to update some documentation about this change

Reply:
No longer relevant since we've removed CA subject inheritance.